Amazon Bedrock support

Amazon Badrock has OpenAI compatible endpoint, but I could not get it to work. Since it is the 'other big' model infering platform it deserves its own connector type which could rely on native authentication and proper model settings/translations.

still the same error 
```
7:40:32 PM [WARN] Unknown model "claude-haiku-4-5-20251001-v1" (from "us.anthropic.claude-haiku-4-5-20251001-v1"), assuming inference profile support
7:40:32 PM [WARN] Bedrock does not support metadata parameter, stripping
7:40:32 PM [INFO] Creating Bedrock credential chain for account bedrock using profile: default
7:40:32 PM [WARN] Validation error: ValidationException
7:40:32 PM [ERROR] Error: The provided model identifier is invalid.. Failing over to next provider.
7:40:32 PM [ERROR] Failed to proxy request with account bedrock
```

this time the packaging seems to be fine, did you fix it?
Bedrock is not there yet:
```
2:47:21 PM [INFO] Request: POST /v1/messages
2:47:21 PM [INFO] Attempting request with account: br (provider: bedrock)
2:47:21 PM [INFO] Creating Bedrock credential chain for account br using profile: default
2:47:21 PM [INFO] Matched client model "claude-haiku-4-5-20251001" to Bedrock model "anthropic.claude-haiku-4-5-20251001-v1:0" (similarity: 100.0%)
2:47:21 PM [INFO] Stripping :0 suffix from model ID: anthropic.claude-haiku-4-5-20251001-v1:0 -> anthropic.claude-haiku-4-5-20251001-v1
2:47:21 PM [INFO] Creating Bedrock credential chain for account br using profile: default
2:47:21 PM [WARN] Unknown model "claude-haiku-4-5-20251001-v1" (from "us.anthropic.claude-haiku-4-5-20251001-v1"), assuming inference profile support
2:47:21 PM [WARN] Bedrock does not support metadata parameter, stripping
2:47:21 PM [INFO] Creating Bedrock credential chain for account br using profile: default
2:47:22 PM [WARN] Validation error: ValidationException
2:47:22 PM [ERROR] Error: The provided model identifier is invalid.. Failing over to next provider.
2:47:22 PM [ERROR] Failed to proxy request with account br
```

still not there:
```
9:35:55 AM [INFO] Request: POST /v1/messages
9:35:55 AM [INFO] Attempting request with account: br (provider: bedrock)
9:35:55 AM [INFO] Creating Bedrock credential chain for account br using profile: default
9:35:55 AM [INFO] Matched client model "claude-haiku-4-5-20251001" to Bedrock model "anthropic.claude-haiku-4-5-20251001-v1:0" (similarity: 100.0%)
9:35:55 AM [INFO] Stripping :0 suffix from model ID: anthropic.claude-haiku-4-5-20251001-v1:0 -> anthropic.claude-haiku-4-5-20251001-v1 9:35:55 AM [INFO]
Creating Bedrock credential chain for account br using profile: default
9:35:55 AM [WARN] Bedrock does not support metadata parameter, stripping
9:35:55 AM [INFO] Creating Bedrock credential chain for account br using profile: default
9:35:55 AM [WARN] Validation error: ValidationException
9:35:55 AM [ERROR] Error: The provided model identifier is invalid.. Failing over to next provider.
9:35:55 AM [ERROR] Failed to proxy request with account br
```

--- a/packages/proxy/src/handlers/proxy-operations.ts
+++ b/packages/proxy/src/handlers/proxy-operations.ts
@@ -11,6 +11,28 @@ import { getValidAccessToken } from "./token-manager";
 const log = new Logger("ProxyOperations");
 
 /**
+ * Bedrock provider currently returns a synthetic Request containing the
+ * provider response payload (instead of a real URL to fetch).
+ * Detect and unwrap that request so we don't try to fetch a fake host.
+ */
+function isSyntheticProviderResponse(request: Request): boolean {
+	return (
+		request.headers.get("x-bedrock-response") === "true" &&
+		request.url.startsWith("https://bedrock.aws/response")
+	);
+}
+
+function materializeSyntheticResponse(request: Request): Response {
+	const headers = new Headers(request.headers);
+	headers.delete("x-bedrock-response");
+
+	return new Response(request.body, {
+		status: 200,
+		headers,
+	});
+}
+
+/**
  * Filters thinking blocks from request body
  * Used when Claude rejects thinking blocks with invalid signatures from other providers
  * @param requestBodyBuffer - The original request body buffer
@@ -327,8 +349,10 @@ export async function proxyWithAccount(
 			? await provider.transformRequestBody(providerRequest, account)
 			: providerRequest;
 
-		// Make the request
-		let rawResponse = await makeProxyRequest(transformedRequest);
+		// Make the request (or unwrap a synthetic provider response)
+		let rawResponse = isSyntheticProviderResponse(transformedRequest)
+			? materializeSyntheticResponse(transformedRequest)
+			: await makeProxyRequest(transformedRequest);
 
 		// Check if this is a Claude provider and we got an invalid thinking signature error
 		const isClaudeProvider =
@@ -359,8 +383,10 @@ export async function proxyWithAccount(
 					? await provider.transformRequestBody(retryProviderRequest, account)
 					: retryProviderRequest;
 
-				// Make the retry request
-				rawResponse = await makeProxyRequest(retryTransformedRequest);
+				// Make the retry request (or unwrap a synthetic provider response)
+				rawResponse = isSyntheticProviderResponse(retryTransformedRequest)
+					? materializeSyntheticResponse(retryTransformedRequest)
+					: await makeProxyRequest(retryTransformedRequest);
 			} else {
 				log.warn(
 					"Failed to filter thinking blocks or no changes made, proceeding with original error response",

